parsers/MD.py

```diff
@@ -129,43 +122,26 @@ def template_exchange_response(
     }
 
 
-def get_archive_data(session: Session | None = None, dates=None) -> list:
-    """
-    Returns archive data as a list of ArchiveDatapoint.
-
-    Optionally accepts a date to fetch the data for,
-    or two dates specifing the range to fetch the data for.
-    Specifying a date-range too high will cause errors with the archive-server.
-    If no dates are specified data for the last 24 hours is fetched.
-    """
-    s = session or Session()
+def get_archive_data(
+    target_datetime: datetime, session: Session | None = None
+) -> list[ArchiveDatapoint]:
+    """Returns archive data for the target day as a list of ArchiveDatapoint."""
 
-    try:
-        date1, date2 = sorted(dates)
-    except Exception:
-        date1 = date2 = dates
-
-    archive_url = archive_base_url
-    if date1 and date2:
-        archive_url += "&date1={}".format(
-            arrow.get(date1).to("Europe/Chisinau").format(archive_date_format)
-        )
-        archive_url += "&date2={}".format(
-            arrow.get(date2).to("Europe/Chisinau").format(archive_date_format)
-        )
+    archive_date = target_datetime.astimezone(TZ).strftime("%d.%m.%Y")
+    archive_url = f"{archive_base_url}&date1={archive_date}&date2={archive_date}"
 
+    s = session or Session()
     data_response = s.get(archive_url, verify=False)
     data = data_response.json()
     try:
-        return [
+        archive_datapoints = [
             ArchiveDatapoint(
-                arrow.get(
-                    entry[0], archive_datetime_format, tzinfo="Europe/Chisinau"
-                ).datetime,
+                datetime.strptime(entry[0], "%Y-%m-%d %H:%M").replace(tzinfo=TZ),
                 *map(float, entry[1:]),
             )
             for entry in data
         ]
+        return sorted(archive_datapoints, key=lambda x: x.datetime)
```

Comment:
Using `itemgetter` instead of lambda is both simpler and faster (there is other benefits as well but are not relevant in our case). So I would recomend that.

But does this need to be sorted?

Reply:
TIL - thanks @VIKTORVAV99 !

The API is/was returning data in reverse chronological order, so production and consumption data are also returned unsorted (at least until we update to event classes). I wasn't too sure of how the backend handles these, but just in case I thought it would be good to sort things properly to limit surprises / keep things intuitive

Reply:
The backend don't care at all which order things are returned in, this is for us humans.